### PR TITLE
Simplify callfunc more

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

**Comment by vgvassilev** [comment]
Depends on #10850

File: core/metacling/src/TClingCallFunc.h [hunk 1]
```diff
@@ -142,22 +142,21 @@ class TClingCallFunc {
    ~TClingCallFunc() = default;
 
    explicit TClingCallFunc(cling::Interpreter *interp)
-      : fInterp(interp), fWrapper(0), fIgnoreExtraArgs(false), fReturnIsRecordType(false)
```
Comment (Axel-Naumann) [inline core/metacling/src/TClingCallFunc.h]: The commit message does not seem to correspond to the code change. I would have expected
```
fReturnIsRecordType is now unused, remove it.
```

Comment (vgvassilev) [reply core/metacling/src/TClingCallFunc.h]: Fixed.

**Comment by vgvassilev** [comment]
Closing in favor of #11252 which is a superset of this PR.

**Pull request closed without merging**